nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -69,6 +70,36 @@ void LatticeMotionTable::initMotionModel(
 
   // TODO(Matt) populate num_angle_quantization, size_x, min_turning_radius, trig_values,
   // all of the member variables of LatticeMotionTable
+
+  std::ifstream latticeFile(current_lattice_filepath);
+
+  if( !latticeFile.is_open() )
+  {
+    throw std::runtime_error("Could not open lattice file");
+  }
+
+  nlohmann::json j;
+  latticeFile >> j;
+
+  float prevStartAngle = 0; 
+  std::vector<MotionPrimitive> primitives; 
+  for(auto const& primative : j["primitives"] )
+  {
+    MotionPrimitive newPrimitive; 
+    fromJsonToMotionPrimitive(primative, newPrimitive); 
+
+    if(prevStartAngle != newPrimitive.startAngle )
+    {
+      motionPrimitives.emplace_back(primitives);
+      primitives.clear(); 
+      prevStartAngle = newPrimitive.startAngle;
+    }
+    primitives.emplace_back(newPrimitive);
+  }
+  motionPrimitives.emplace_back(primitives); 
+ 
+  fromJsonToMetaData(j["latticeMetadata"], latticeMetadata);
```

Comment:
Lets move the metadata first since in the file its also first (and I think generally it helps color the code to understand what's happening under it if we deal with the header first organizationally)

Reply:
Done